configure.ac

```diff
@@ -1144,12 +1142,26 @@ fi
 if test "$with_hwloc" = "embedded" ; then
     PAC_PUSH_FLAG([CFLAGS])
     CFLAGS="$VISIBILITY_CFLAGS"
-    PAC_CONFIG_SUBDIR_ARGS([modules/hwloc], [--enable-embedded-mode --disable-visibility],[], [AC_MSG_ERROR(embedded hwloc configure failed)])
+    hwloc_config_args="--enable-embedded-mode --disable-visibility"
+    hwloc_config_args="$hwloc_config_args --disable-libxml2"
+    hwloc_config_args="$hwloc_config_args --disable-nvml"
+    hwloc_config_args="$hwloc_config_args --disable-cuda"
+    hwloc_config_args="$hwloc_config_args --disable-opencl"
+    hwloc_config_args="$hwloc_config_args --disable-rsmi"
+    PAC_CONFIG_SUBDIR_ARGS([modules/hwloc], [$hwloc_config_args],[], [AC_MSG_ERROR(embedded hwloc configure failed)])
     PAC_POP_FLAG([CFLAGS])
     hwlocsrcdir="${main_top_builddir}/modules/hwloc"
     hwloclib="${main_top_builddir}/modules/hwloc/hwloc/libhwloc_embedded.la"
     PAC_APPEND_FLAG([-I${use_top_srcdir}/modules/hwloc/include],[CPPFLAGS])
     PAC_APPEND_FLAG([-I${main_top_builddir}/modules/hwloc/include],[CPPFLAGS])
+
+    # capture the line -- S["HWLOC_EMBEDDED_LIBS"]="-lm "
+    hwloc_embedded_libs=$(awk -F'"' '/^S."HWLOC_EMBEDDED_LIBS"/ {print $4}' modules/hwloc/config.status)
```

Comment:
Oh, maybe we should use `AC_PROG_AWK` and $AWK here in case the system has `gawk` or something.

Reply:
Oops, I merged it before addressing this comment.  In practice, I believe `awk` is always available. Systems often create symbolic links in the case of `gawk` or `nawk`. I think this is similar to `sed` where we seem always directly call `sed` rather than `$SED` as long as we don't use extended features.